master/buildbot/reporters/mail.py

```diff
@@ -43,6 +43,11 @@
 from buildbot.reporters.message import MessageFormatter as DefaultMessageFormatter
 from buildbot.util import service
 
+if PY3:
```

Comment:
@tardyp sorry I am not able to attend the buildbot meeting.

The places where I used StringIO all interact with sys.stdin and sys.stdio.
Most of these instances are located in tests.
These are implemented differently on Python 2 and Python 3, where on Python 2
they are bytes, and and Python 3, there are unicode.

As @edschofield has [pointed out](https://github.com/buildbot/buildbot/pull/2429#issuecomment-260717355)
So, some logic around "if PY3" *is* required to get these tests working on Python 2 and 3,
since sys.stdin and sys.stdout are different on Python 2 and Python 3.

What I have done does work on Python 2 and Python 3.  If you don't like that approach
what do you want to do instead?